code/Serving_API.py

```diff
@@ -1,27 +1,258 @@
 #!/usr/bin/env python3
 
 from fastapi import FastAPI
+from ConvertToStandardPath_MargeSubmarineWithLandCable import get_all_submarine_to_standard_paths_pairs
+from ConvertToStandardPath_SubmarineCable import get_all_submarine_standard_paths, floatFormatter
+import sqlite3
+import networkx as nx
+import csv
+from xml.dom.minidom import Document
+from haversine import haversine
+
+# find the closest point for a given coordinate in case it is not in the graph
+
+
+def find_closest_point(point, points_set):
+    return min(points_set, key=lambda p: haversine(point, p))
+
+# helper function to build nx graph with src/dst city, src/dst coordinates, wkt path, cabel type and distance.
+
+
+def add_edge(G, city1, city2, distance, path_wkt, src_city_coord, dst_city_coord, cable_type):
+    # Add or update nodes with their coordinates
+    G.add_node(city1, coord=src_city_coord)
+    G.add_node(city2, coord=dst_city_coord)
+
+    # Add the edge with its properties
+    G.add_edge(city1, city2, weight=distance, path_wkt=path_wkt,
+               src_city_coord=src_city_coord, dst_city_coord=dst_city_coord, cable_type=cable_type)
+
+# read standard paths from database
+
+
+def get_all_standard_paths(db_file):
+    conn = sqlite3.connect(db_file)
+    cursor = conn.cursor()
+    sql_query = """
+    SELECT sp.from_city, sp.from_state, sp.from_country, sp.to_city, sp.to_state, sp.to_country, sp.distance_km, sp.path_wkt
+    FROM standard_paths sp
+    """
+    cursor.execute(sql_query)
+    datas = cursor.fetchall()
+    conn.close()
+    return datas
+
+# lati and longti in wkt path is reversed, use this helper function to reverse that
+
+
+def coordinate_reverser(coord):
+    coord_longti, coord_lati = coord
+    return (coord_lati, coord_longti)
+
+# reverse linestring coords order for direct graph
+
+
+def reverse_linestring_order(linestring):
```

Comment:
I'd suggest load the string using wkt.loads, reverse the coordinates and then output
e.g.
```Python
    from shapely import wkt
    from shapely.geometry import LineString
    ls = wkt.loads(linestring)
    ls_reversed = LineString(coordinates[::-1])
    return ls_reversed.wkt
```